tools/mascot/src/main/java/dk/alexandra/fresco/tools/mascot/MascotResourcePoolImpl.java

```diff
@@ -0,0 +1,144 @@
+package dk.alexandra.fresco.tools.mascot;
+
+import dk.alexandra.fresco.commitment.HashBasedCommitment;
+import dk.alexandra.fresco.commitment.HashBasedCommitmentSerializer;
+import dk.alexandra.fresco.framework.network.Network;
+import dk.alexandra.fresco.framework.network.serializers.ByteSerializer;
+import dk.alexandra.fresco.framework.network.serializers.StrictBitVectorSerializer;
+import dk.alexandra.fresco.framework.sce.resources.ResourcePoolImpl;
+import dk.alexandra.fresco.framework.util.Drbg;
+import dk.alexandra.fresco.framework.util.ExceptionConverter;
+import dk.alexandra.fresco.framework.util.StrictBitVector;
+import dk.alexandra.fresco.tools.cointossing.CoinTossing;
+import dk.alexandra.fresco.tools.mascot.field.FieldElementSerializer;
+import dk.alexandra.fresco.tools.mascot.utils.FieldElementPrg;
+import dk.alexandra.fresco.tools.mascot.utils.FieldElementPrgImpl;
+import dk.alexandra.fresco.tools.ot.base.RotBatch;
+import dk.alexandra.fresco.tools.ot.otextension.BristolRotBatch;
+import dk.alexandra.fresco.tools.ot.otextension.RotList;
+import dk.alexandra.fresco.tools.ot.otextension.OtExtensionResourcePool;
+import dk.alexandra.fresco.tools.ot.otextension.OtExtensionResourcePoolImpl;
+import java.math.BigInteger;
+import java.security.MessageDigest;
+import java.util.List;
+import java.util.Map;
+
+public class MascotResourcePoolImpl extends ResourcePoolImpl implements MascotResourcePool {
+
+  private final Map<Integer, RotList> seedOts;
+  private final List<Integer> partyIds;
+  private final int instanceId;
+  private final BigInteger modulus;
+  private final int modBitLength;
+  private final int lambdaSecurityParam;
+  private final int prgSeedLength;
+  private final int numLeftFactors;
+  private final FieldElementPrg localSampler;
+  private final FieldElementSerializer fieldElementSerializer;
+  private final StrictBitVectorSerializer strictBitVectorSerializer;
+  private final ByteSerializer<HashBasedCommitment> commitmentSerializer;
+  private final MessageDigest messageDigest;
+
+  /**
+   * Creates new mascot resource pool.
+   */
+  public MascotResourcePoolImpl(Integer myId, List<Integer> partyIds,
```

Comment:
Please document the various arguments. Considering the number of arguments it may make sense to use a different strategy for constructing these objects, e.g., a Builder.